### DISCUSSION TITLE
ONVIF support

### DISCUSSION THREAD

**Discussion by ghost** [mrlt8/docker-wyze-bridge#727] [Ideas]
Nice work!

I would like to integrate this into Samsung Smarthings.
Is it possible to ONVIF to the bridge?

-csk

**Comment by mrlt8** [comment]
I think ONVIF would be great, but I don't believe there are any python libraries available to serve ONVIF/SOAP.
Open to suggestion or examples of a SOAP server.

**Comment by ghost** [comment]
Thankyou for your reply.

Does this help?

https://elsampsa.github.io/valkka-examples/_build/html/onvif.html

**Reply by mrlt8** [reply]
Thanks, but I believe that is for connecting to an OnVif device?

**Comment by ghost** [comment]
Thankyou for the followup.

My thought is implementing ONVIF protocol without your bridge so the WYZE cams act as being ONVIF compliant.

-Chet

**Comment by joeblack2k** [comment]
https://github.com/FalkTannhaeuser/python-onvif-zeep 

isnt this a library?

**Reply by mrlt8** [reply]
I believe that's mostly to connect to an existing onvif camera, whereas we would be trying to do the reverse and emulate an onvif camera/server.

**Comment by jpriganc** [comment]
@mrlt8  - I know this is an old thread, but I wanted to check to see if this might be possible now? I found the following repository and thought it might be what you needed to get onvif integrated. With Ubiquiti opening up their Protect line to onvif supported devices, I could see this as a great use case! Thanks for the work on maintaining this!

https://github.com/IroN404/onvif-camera-mock

**Reply by idaband** [reply]
i saw that same project but was a bit confused.... did you get this to work at all. 

I'm in the same boat.  Want to use wyze in Unifi Protect now too.

**Reply by mrlt8** [reply]
Some basic support for onvif on #1355 and the `wyze-bridge:onvif` image.

**Reply by idaband** [reply]
trying it out.  Didn't see the port /ect for the onvif stream in the code.  

Unifi just looks for an IP address with port.  Not sure they can take host with path yet.  ...they might also require a username/password to "adopt" the camera too.

**Reply by bdog4life** [reply]
Curious about this as well - If it can be done, I would use this

**Reply by waterboy300** [reply]
Just wanted to bump this and see if there are any updates

**Reply by ePoCShadowz** [reply]
I have got it working, Wrote an OnVif intermediary docker container in python and have 3 wyze cam v4s in Unifi Protect, 
Will not make it a public project but would be happy to share the code with those interested in adapting it

**Comment by joleaj** [comment]
Hi ePoCShadowz,

I’m reaching out after seeing your GitHub comment about your ONVIF
intermediary container for Wyze Cam V4s. I’m currently in a similar
situation — I have two Wyze Window Cams (model GW_WC) and am trying to get
them working in Home Assistant/Frigate via ONVIF, since the
docker-wyze-bridge SDK doesn’t yet support my camera model (I’m getting
IOTC_ER_UNLICENSE).

Your solution sounds like exactly what I need! I’d be incredibly grateful
if you could share your code with me — I’d love to adapt it for my setup
and potentially contribute improvements back if I can.

I’m running Home Assistant OS on an HP laptop and have Frigate configured
to pull streams. I’m also working on a custom dashboard, so being able to
feed live video into that would be a game-changer.

Thank you for your work and for being open to sharing — I truly appreciate
it!

Jolea
